BepInEx.IL2CPP/PluginClassInjector.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Reflection;
+using UnhollowerBaseLib;
+using UnhollowerRuntimeLib;
+
+namespace BepInEx.IL2CPP
+{
+
+    [AttributeUsage(AttributeTargets.Class)]
+    public class Il2CppInterfaces : Attribute
+    {
+
+        public Type[] Interfaces { get; protected set; }
+
+        public Il2CppInterfaces(params Type[] interfaces)
+        {
+            Interfaces = interfaces;
+        }
+
+    }
+
+    [AttributeUsage(AttributeTargets.Assembly | AttributeTargets.Class)]
+    public class DontAutoRegisterInIl2Cpp : Attribute { }
+
+    internal class PluginClassInjector
+    {
+
+        public static void RegisterAssemblyInIl2Cpp(Assembly pluginAssembly)
+        {
+            if (pluginAssembly.GetCustomAttribute<DontAutoRegisterInIl2Cpp>() != null) return;
+
+            foreach (var type in pluginAssembly.DefinedTypes)
+                if (typeof(Il2CppObjectBase).IsAssignableFrom(type) && !ClassInjector.IsTypeRegisteredInIl2Cpp(type))
+                    RegisterTypeInIl2Cpp(type);
+        }
+
+        public static bool RegisterTypeInIl2Cpp(Type type)
```

Comment:
The bool indicates if the type was registered.
Have a look at line 42-44 that's how the base Types are loaded and if it doesn't load, then the class which inherits will not load aswell to prevent errors.